synapse/lib/cell.py

```diff
@@ -4932,3 +4943,41 @@ def getCachedSslCtx(self, opts=None, verify=None):
 
         key = tuple(sorted(opts.items()))
         return self._sslctx_cache.get(key)
+
+    async def freeze(self, timeout=30):
+
+        if self.paused:
+            mesg = 'The service is already frozen.'
+            raise s_exc.BadState(mesg=mesg)
+
+        logger.warning(f'Freezing service for volume snapshot.')
+
+        logger.warning('...acquiring nexus lock to prevent edits.')
+        await asyncio.wait_for(self.nexslock.acquire(), timeout=timeout)
```

Comment:
Do we want to try/catch the `asyncio.Timeout` and throw a `s_exc.TimeOut` where we can give a meaningful message to the caller?